### ISSUE TITLE
[Question] Exposition and what is it eaten with

### ISSUE THREAD

**Issue by Extegral** [Vazkii/Patchouli#303]
I've been checking out on some of the pull requests here, and it seems like mysterious phenomenon called "exposition" if often mentioned in debates related to API changes, mostly in negative context. I have of course tried googling it up first, but information I've managed to find is pretty generic and doesn't provide an image of how it relates to Patchouli and why is it all that much important here. Whatever this thing is, it seems to have had zero relevance to my former modding experience, so I really can't guess the latter.

It would be nice to have a detailed explanation on this matter. I assume that this topic might be important for people who'd like to contribute to this mod's development in future, so I have opened an issue for the sake of proper archival.

**Comment by williewillus** [comment]
It's not a general term. What I mean is you have to be careful with what you expose to users via the API, because you can't take that back, at least not until the next Minecraft version. Even then, people prefer the API's they use to remain stable and not shift around all the time.

So the standard for letting people outside Patchouli use something via the API is much higher than normal implementation inside the mod, and requires careful planning and design.

**Comment by Extegral** [comment]
Well, makes sense now that I think about it. I don't have any other questions on this so far.
